bitnami/owncloud/templates/tls-secrets.yaml

```diff
@@ -0,0 +1,43 @@
+{{- if .Values.ingress.enabled }}
```

Comment:
I think this won't work for multiple hosts, since multiple ingress entries with the same name would be created (and only the last would be taken into account).

A user reported an issue with this format, see #4618 for the issue and #4707 for the fix. If you copy it from a chart like Drupal it would work.

Reply:
I don't think issue applies to this case @marcosbc 
See this example:
```console
$ helm template owncloud bitnami/owncloud \
    --set ingress.enabled=true \
    # Main domain
    --set ingress.tls=true --set ingress.hostname=main.domain \
    # Extra Domains
    --set ingress.extraHosts[0].name=second.domain --set ingress.extraHosts[0].path=/ \
    --set ingress.extraTls[0].hosts[0]=second.domain --set ingress.extraTls[0].secretName=second.domain-tls \
    --set ingress.secrets[0].name=second.domain-tls --set ingress.secrets[0].key="PRIVATE KEY ..." --set ingress.secrets[0].certificate="BEGIN ..." \
    -s templates/ingress.yaml -s templates/tls-secrets.yaml

---
# Source: owncloud/templates/ingress.yaml
apiVersion: networking.k8s.io/v1beta1
kind: Ingress
metadata:
  name: owncloud
  namespace: default
  labels:
    app.kubernetes.io/name: owncloud
    helm.sh/chart: owncloud-10.0.0
    app.kubernetes.io/instance: owncloud
    app.kubernetes.io/managed-by: Helm
spec:
  rules:
    - host: main.domain
      http:
        paths:
          - path: /
            backend:
              serviceName: owncloud
              servicePort: http
    - host: second.domain
      http:
        paths:
          - path: /
            backend:
              serviceName: owncloud
              servicePort: http
  tls:
    - hosts:
        - main.domain
      secretName: main.domain-tls
    - hosts:
        - second.domain
      secretName: second-secret
---
# Source: owncloud/templates/tls-secrets.yaml
apiVersion: v1
kind: Secret
metadata:
  name: second-secret
  namespace: default
  labels:
    app.kubernetes.io/name: owncloud
    helm.sh/chart: owncloud-10.0.0
    app.kubernetes.io/instance: owncloud
    app.kubernetes.io/managed-by: Helm
type: kubernetes.io/tls
data:
  tls.crt: "..."
  tls.key: "..."
# Source: owncloud/templates/tls-secrets.yaml
apiVersion: v1
kind: Secret
metadata:
  name: main.domain-tls
  namespace: default
  labels:
    app.kubernetes.io/name: owncloud
    helm.sh/chart: owncloud-10.0.0
    app.kubernetes.io/instance: owncloud
    app.kubernetes.io/managed-by: Helm
type: kubernetes.io/tls
data:
  tls.crt: "..."
  tls.key: "..."
  ca.crt: "..."
```
As you can see, I'm able to create a default ingress rule + tls configuration for "main" doman, and add an extra one for a second domain.

Reply:
Ok then, in that case my only concern is regarding consistency, since they are both PHP applications and TLS secrets config should be the same for both. If this is the "correct" approach then we should probably fix the others as well.